### ISSUE TITLE
Issue on weight decay

### ISSUE THREAD

**Issue by liujingcs** [davda54/sam#52]
Thanks for a great job. It seems the first step does not consider the effect of weight decay, which is quite different from the TensorFlow version. Does this have a negative effect on the performance?
